mypy/build.py

```diff
@@ -998,6 +999,117 @@ def generate_deps_for_cache(manager: BuildManager,
     return rdeps
 
 
+def save_baseline(manager: BuildManager, sources: List[BuildSource]) -> None:
+    if not manager.options.baseline_file or (
+        (not manager.options.write_baseline and not manager.options.auto_baseline)
+        or (not manager.options.write_baseline and manager.options.auto_baseline and (
+            not manager.errors.num_messages()
+            or manager.errors.baseline_sources != [source.path for source in sources]
+        ))
+    ):
+        # Indicate that writing was canceled
+        manager.options.write_baseline = False
+        return
+    new_baseline = manager.errors.prepare_baseline_errors()
+    file = Path(manager.options.baseline_file)
+    if not new_baseline:
+        if file.exists():
+            file.unlink()
+            print("No errors, baseline file removed")
+        else:
+            print("No errors, no baseline to write")
+        # Indicate that writing was canceled
+        manager.options.write_baseline = False
+        return
+    if new_baseline == manager.errors.original_baseline:
+        return
+    if not file.parent.exists():
+        file.parent.mkdir(parents=True)
+    data: Mapping[str, object]
+    if (
+        manager.options.baseline_format == "1.3"
+        or manager.options.baseline_format == "default"
+    ):
+        data = {
+            "files": new_baseline,
+            "format": "1.3",
+            "sources": [source.path for source in sources],
+        }
+    elif manager.options.baseline_format == "1.2":
+        data = {
+            "__baseline_metadata__": {
+                "format": "1.2", "sources": [source.path for source in sources]
+            },
+            **new_baseline,
+        }
+    else:
+        raise Exception(
+            f"Invalid/Unsupported baseline file format {manager.options.baseline_format}"
+        )
+    json.dump(
+        data,
+        file.open("w"),
+        indent=2,
+        sort_keys=True,
+    )
+
+
+def load_baseline(options: Options, errors: Errors, stdout: TextIO) -> None:
+    if not options.baseline_file:
+        return
+
+    from mypy import main
+
+    formatter = util.FancyFormatter(stdout, stderr, options.show_error_codes)
+    file = Path(options.baseline_file)
+
+    if not file.exists():
+        if options.baseline_file != defaults.BASELINE_FILE:
+            msg = formatter.style(
+                f"error: Baseline file {options.baseline_file!r} not found", 'red',
+                bold=True
+            )
+            main.fail(msg, stderr, options)
+        return
+    try:
+        data: Dict[str, object] = json.load(file.open())
+    except JSONDecodeError:
+        msg = formatter.style(
+            f"error: Baseline file {options.baseline_file!r} contains invalid JSON",
+            'red', bold=True
+        )
+        main.fail(msg, stderr, options)
+    # try to detect format:
+    format_ = data.get("format")
```

Comment:
why not `format`?

Reply:
shadows the builtin `format`

Reply:
is there a more descriptive name it can use or nah